cmd/sfncli/runner.go

```diff
@@ -20,36 +20,48 @@ const maxTaskOutputLength = 32768
 const maxTaskFailureCauseLength = 32768
 
 type TaskRunner struct {
-	sfnapi    sfniface.SFNAPI
-	taskToken string
-	cmd       string
-	args      []string
+	sfnapi        sfniface.SFNAPI
+	taskToken     string
+	cmd           string
+	args          []string
+	executionName string
 }
 
-func NewTaskRunner(cmd string, args []string, sfnapi sfniface.SFNAPI, taskInput string, taskToken string) TaskRunner {
-	// if the task input is an array of strings, interpret these as an args array
-	// otherwise pass the raw input as a single arg
-	var taskInputArgs []string
-	if err := json.Unmarshal([]byte(taskInput), &taskInputArgs); err != nil {
-		taskInputArgs = []string{taskInput}
+func NewTaskRunner(
+	cmd string, args []string, sfnapi sfniface.SFNAPI, input, taskToken string,
+) (TaskRunner, error) {
+	var job map[string]interface{}
+	if err := json.Unmarshal([]byte(input), &job); err != nil {
+		return TaskRunner{}, fmt.Errorf("Input must be a json object: %s", err)
 	}
 
-	return TaskRunner{
-		sfnapi:    sfnapi,
-		taskToken: taskToken,
-		cmd:       cmd,
-		args:      append(args, taskInputArgs...),
+	executionName, ok := job["$execution-name"].(string)
+	if !ok {
+		return TaskRunner{}, fmt.Errorf("Job with unknown execution-name")
+	}
+
+	marshaledJob, err := json.Marshal(job)
+	if err != nil {
+		return TaskRunner{}, err
 	}
+
+	return TaskRunner{
+		sfnapi:        sfnapi,
+		taskToken:     taskToken,
+		executionName: executionName,
+		cmd:           cmd,
+		args:          append(args, string(marshaledJob)),
+	}, nil
 }
 
 // Process runs the underlying cmd with the appropriate
 // environment and command line params
 func (t TaskRunner) Process(ctx context.Context) error {
 	if t.sfnapi == nil {
-		return nil // if New failed :-/
+		return fmt.Errorf("NewTaskFailure -- nil sfnapi") // if New failed :-/
 	}
 	cmd := exec.CommandContext(ctx, t.cmd, t.args...)
-	cmd.Env = os.Environ()
+	cmd.Env = append(os.Environ(), "_EXECUTION_NAME="+t.executionName)
```

Comment:
`_`s here make me feel like `_`s in the JSON, e.g. `_execution_name` would be more consistent

Reply:
Renamed in 83ee42e